src/domain/usecases/ClosePatientsUseCase.ts

```diff
@@ -0,0 +1,10 @@
+import { Async } from "domain/entities/Async";
+import { ProgramsRepository, ClosePatientsOptions } from "domain/repositories/ProgramsRepository";
+
+export class ClosePatientsUseCase {
+    constructor(private programsRepository: ProgramsRepository) {}
+
+    async execute(options: ClosePatientsOptions): Async<void> {
+        await this.programsRepository.closePatients(options);
```

Comment:
Ok, this will seem to contradict my previous notes, but it's more of an orthogonal issue. A key question on Clean Arch is where the logic should live. If we create a `closePatients` in the repo, it's clear all the logic will live there. I know we do that on some repos in d2-tools, but it should be the last resort. The logic should live in use cases and entities. Repositories should be as stupid as possible. get/create/save/delete, this kind of thing. Assess if we could have  `get/save` and move all other `closePatient` code to the use case.